src/views/News/index.js

```diff
@@ -43,465 +70,556 @@ const styles = {
-      filteredNews: [],
-      network: 'online',
-      newPostCategory: '',
-      newPostSubject: '',
-      newPostBody: '',
-      snackbarOpen: false,
-      snackbarMessage: 'Something went wrong',
-      currentUsername: '',
-      // false if not editing, newsID if editing
-      currentlyEditing: false,
+const StudentNews = (props) => {
+  const [search, setSearch] = useState('');
+  const [openPostActivity, setOpenPostActivity] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [categories, setCategories] = useState([]);
+  const [news, setNews] = useState([]);
+  const [personalUnapprovedNews, setPersonalUnapprovedNews] = useState([]);
+  //const [filteredNews, setFilteredNews] = useState([]);
+  const isOnline = useNetworkStatus();
+  const [newPostCategory, setNewPostCategory] = useState('');
+  const [newPostSubject, setNewPostSubject] = useState('');
+  const [newPostBody, setNewPostBody] = useState('');
+  const [cropperImageData, setCropperImageData] = useState(null); //null if no picture chosen, else contains picture
+  const [photoDialogErrorTimeout, setPhotoDialogErrorTimeout] = useState(null);
+  const [photoDialogError, setPhotoDialogError] = useState(null);
+  const [aspectRatio, setAspectRatio] = useState(null);
+  const [snackbar, setSnackbar] = useState({ open: false, text: '', severity: '' });
+  const [currentUsername, setCurrentUsername] = useState('');
+  const [currentlyEditing, setCurrentlyEditing] = useState(false); // false if not editing, newsID if editing
+
+  let cropperRef = React.createRef();
+
+  const loadNews = useCallback(async () => {
+    setLoading(true);
+    if (props.authentication) {
+      const newsCategories = await newsService.getCategories();
+      const personalUnapprovedNews = await newsService.getPersonalUnapprovedFormatted();
+      const unexpiredNews = await newsService.getNotExpiredFormatted();
+      setLoading(false);
+      setCategories(newsCategories);
+      setNews(unexpiredNews);
+      setPersonalUnapprovedNews(personalUnapprovedNews);
+      //setFilteredNews(unexpiredNews);
+    } else {
+      // TODO: test authentication handling and neaten code (ex. below)
+      // alert("Please sign in to access student news");
+    }
+  }, [props.authentication]);
+
+  useEffect(() => {
+    loadNews();
+  }, [props.authentication, loadNews]);
+
+  useEffect(() => {
+    const loadUsername = async () => {
+      const user = await userService.getProfileInfo();
+      setCurrentUsername(user.AD_Username);
     };
-    this.isMobileView = false;
-    this.breakpointWidth = 540;
-    this.updateSnackbar = this.updateSnackbar.bind(this);
-    this.handleNewsItemEdit = this.handleNewsItemEdit.bind(this);
-    this.callFunction = this.callFunction.bind(this);
-  }
 
-  componentDidMount() {
-    this.setState({ loading: false });
-    this.loadNews();
-    this.loadUsername();
-    window.addEventListener('resize', this.resize);
+    loadUsername();
+  });
```

Comment:
This `loadUsername()` effect is being run on every render. Can we add an empty dependency list so it only depends on the first render?